internal/db/rdb/mysql/mysql_test.go

```diff
@@ -151,80 +186,172 @@ func Test_mySQLClient_Open(t *testing.T) {
+					maxOpenConns:         100,
+					maxIdleConns:         100,
+					session: &dbr.MockSession{
+						PingContextFunc: func(ctx context.Context) error {
+							return nil
+						},
+					},
+					connected: func() atomic.Value {
+						v := new(atomic.Value)
+						v.Store(false)
+						return *v
+					}(),
+					dbr: &dbr.MockDBR{
+						OpenFunc: func(driver, dsn string, log EventReceiver) (dbr.Connection, error) {
+							conn := &dbr.MockConn{
+								NewSessionFunc: func(event EventReceiver) dbr.Session {
+									return &dbr.MockSession{
+										PingContextFunc: func(ctx context.Context) error {
+											return nil
+										},
+									}
+								},
+								SetConnMaxLifetimeFunc: func(d time.Duration) {},
+								SetMaxIdleConnsFunc:    func(n int) {},
+								SetMaxOpenConnsFunc:    func(n int) {},
+							}
+							return conn, nil
+						},
+					},
+				},
+				want: want{},
+				afterFunc: func(args) {
+					go func() {
+						cancel()
+					}()
+				},
+			}
+		}(),
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			dialer, _ := tcp.NewDialer()
+			dialerFunc := dialer.GetDialer()
+			return test{
+				name: "Open success with dialer when no error occurs",
+				args: args{
+					ctx: ctx,
+				},
+				fields: fields{
+					db:                   "vdaas",
+					host:                 "vald.com",
+					port:                 3306,
+					user:                 "vdaas",
+					pass:                 "vald",
+					name:                 "vald-user",
+					charset:              "utf8bm4j",
+					timezone:             "Local",
+					initialPingTimeLimit: 1000 * time.Microsecond,
+					initialPingDuration:  10 * time.Microsecond,
+					connMaxLifeTime:      1 * time.Microsecond,
+					dialer:               dialer,
+					dialerFunc:           dialerFunc,
+					maxOpenConns:         100,
+					maxIdleConns:         100,
+					session: &dbr.MockSession{
+						PingContextFunc: func(ctx context.Context) error {
+							return nil
+						},
+					},
+					connected: func() atomic.Value {
+						v := new(atomic.Value)
+						v.Store(false)
+						return *v
+					}(),
+					dbr: &dbr.MockDBR{
+						OpenFunc: func(driver, dsn string, log EventReceiver) (dbr.Connection, error) {
+							conn := &dbr.MockConn{
+								NewSessionFunc: func(event EventReceiver) dbr.Session {
+									return &dbr.MockSession{
+										PingContextFunc: func(ctx context.Context) error {
+											return nil
+										},
+									}
+								},
+								SetConnMaxLifetimeFunc: func(d time.Duration) {},
+								SetMaxIdleConnsFunc:    func(n int) {},
+								SetMaxOpenConnsFunc:    func(n int) {},
+							}
+							return conn, nil
+						},
+					},
+				},
+				want: want{},
+				afterFunc: func(args) {
+					go func() {
+						cancel()
+					}()
+				},
+			}
+		}(),
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			return test{
+				name: "returns error when dbr.Open failed",
```

Comment:
can you also add a test case of ping error?

Reply:
I'm not sure why add this pattern.
ping error test is created in Ping func test.
could you please show me your opinion?

Reply:
My idea was to think about how `dbr.Open()` works. (like black box testing and white box testing)
In our implementation, `dbr.Open()` will ping the db and return the error, but in the test code there are no such test case.

If you think it is ok to not to implement such test case, I think it is ok too